ruby_enumerable_methods.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    for item in self do
+      yield(item)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    for item in self do
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    select_arr = []
+    for item in self do
+      select_arr << item if yield(item)
+    end
+    select_arr
+  end
+
+  def my_all?
+    pass = false
+    for item in self do
+      pass = yield(item)
+      break unless pass
+    end
+    pass
+  end
+
+  def my_any?
+    pass = false
+    for item in self do
+      pass = yield(item)
+      break if pass
+    end
+    pass
+  end
+
+  def my_none?
+    pass = false
+    for item in self do
+      pass = !yield(item)
+      break unless pass
+    end
+    pass
+  end
+
+  def my_count
+    length
+  end
+
+  def my_map(my_proc = nil)
+    new_arr = []
+    return self unless my_proc || block_given?
+
+    for item in self do
+      block_given? ? new_arr << yield(item) : new_arr << code.call(item)
+    end
+    new_arr
+  end
```

Comment:
- enumerables my_map returns an Enumerator if no block is given
      Failure/Error: expect(array.my_map).to be_an(Enumerator)